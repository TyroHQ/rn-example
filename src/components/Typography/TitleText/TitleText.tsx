import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";

const TitleTextContainer = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class TitleText extends React.Component<{}> {
  static defaultProps: TitleText["props"];
  render() {
    return <TitleTextContainer>Title text</TitleTextContainer>;
  }
}
