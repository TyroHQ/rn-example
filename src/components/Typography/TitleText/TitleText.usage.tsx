import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { TitleText } from "./TitleText";

export const basic = () => <TitleText />;
