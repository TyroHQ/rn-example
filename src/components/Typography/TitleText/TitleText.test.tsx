import { shallow } from "enzyme";
import * as usage from "./TitleText.usage";

describe("<TitleText />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
