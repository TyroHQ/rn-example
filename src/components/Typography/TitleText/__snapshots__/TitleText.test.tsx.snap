// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleText /> renders well in basic mode 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  Title text
</Text>
`;
